src/persist-client/src/internal/machine.rs

```diff
@@ -79,9 +80,18 @@ where
         shard_id: ShardId,
         metrics: Arc<Metrics>,
         state_versions: Arc<StateVersions>,
-        shared_states: &StateCache,
+        shared_states: &Arc<StateCache>,
```

Comment:
since this unconditionally clones the statecache, let's make it take an owned version and push the clone up a level of the call stack